core/CMakeLists.txt

```diff
@@ -40,9 +40,7 @@ catkin_package(
 )
 
 add_compile_options(-fvisibility-inlines-hidden)
-if(ENABLE_PRUNING)
-	add_compile_definitions(ENABLE_PRUNING)
-endif()
+add_compile_definitions(ENABLE_PRUNING=$<BOOL:${ENABLE_PRUNING}>)
```

Comment:
What is that doing?

Reply:
This is exposing the cmake option as a compiler definition to serve as the default of `setPruning()` below.